[TECH] Migration vers @nuxt/image (PIX-3945)

yannbertrand opened the pull request:
## :unicorn: Problème
Il est conseillé de préférer l'utilisation de `<nuxt-img>` (`@nuxt/image`) à la place de `<prismic-image>`. Cela permet entre autres l'optimisation automatisée des images. Exemple : si les `width` et `height` sont précisées, les images sont retaillées à la compilation pour ne pas prendre l'image originale.

Voir https://prismic.io/blog/nuxt-image-is-out-so-is-its-prismic-integration pour plus d'infos.

## :robot: Solution
Installation du package `@nuxt/image`  et utilisation de la balise `<nuxt-img>`.

## :rainbow: Remarques
En dev, les images viennent directement des CDN Prismic.

Il a fallu modifier la manière dont on calculait le nombre de colonnes (en JS) car @nuxt/image n'est pas capable de détecter les images téléchargées côté client (`mounted` + `process.client`). C'est maintenant fait en CSS, j'ai pris la liberté de ne gérer plus que 2 layout (mobile et large-screen) plutôt que 4 auparavant.

## :100: Pour tester
Vérifier dans la RA que les images sont toujours affichées, qu'elles sont bien rapatriées depuis Prismic (= hebergées sur pix-site = les images ont des `src` == `/_nuxt/images/...`) et que celles qui ont les attributs `width` et `height` ont bien les dimensions souhaitées (elles doivent être redimensionnées à la compil).

Vérifier que les layouts sont acceptables en desktop ET en mobile. Vérifier aussi que les tailles des images sont correctes.

yannbertrand committed:
feat: migrate to @nuxt/image

pix-service commented:
I'm deploying this PR to these urls:

- Pix Site (fr): https://site-pr327.review.pix.fr
- Pix Site (org): https://site-pr327.review.pix.org
- Pix Pro (fr): https://pro-pr327.review.pix.fr

 Please check it out!

yannbertrand committed:
fix: use static assets

MelanieMEB reviewed (approved):
Ca pourrait en + régler un problème d'alt avec le prismic-image (pas de alt coté prismic = pas de alt coté code plutot que le `alt=""`)

yannbertrand committed:
fix: auto add image width & height attributes

Use image dimensions by default.

If the `maxHeight` prop is given and the image height is too big, use the `maxHeight` and compute the `width` accordingly.

Caution: image sizes could be different from what they were before.

yannbertrand committed:
feat: use pix-image  new attribute

yannbertrand committed:
fix: do not resize article images

yannbertrand committed:
feat: add a param to avoid resizing responsive images

yannbertrand committed:
feat: remove assets-extractor-module which is not necessary anymore

yannbertrand committed:
feat: use CSS for MultipleBlock grid layout rather than JS

Nuxt image can't resolve images that are downloaded client side. So using `mounted` which is not executed in Nuxt server lifecycle + using `process.client` is not possible.
I tried to migrate the layout calculation from JS to CSS using flexbox.

yannbertrand committed:
feat: download page banner background image during static build
